mods/ra/maps/allies-01/allies01.lua

```diff
@@ -25,7 +25,7 @@ SendInsertionHelicopter = function()
 	local tanya = passengers[1]
 	Trigger.OnKilled(tanya, TanyaKilledInAction)
 	Trigger.AfterDelay(DateTime.Seconds(4), function()
-		Media.DisplayMessage("According to the rules of engagement I need your explicit orders to fire, Commander!", "Tanya")
+		Media.DisplayMessage(UserInterface.Translate("tanya-rules-of-engagement"), UserInterface.Translate("tanya"))
```

Comment:
IMO `Media.DisplayMessage` should support passing the translation ID like other methods instead of requiring a pre-translated message.

Reply:
We decided against it in the previous pull requests to make it more visible and to avoid unnecessary translation lookups for thirdparty scripts that are not localized.

Reply:
Can you please link to the previous discussion so I can get this context?

Reply:
https://github.com/OpenRA/OpenRA/pull/20265

Reply:
All i could find was one comment about the default value for the default objective type.

The rest of our code (e.g. chrome logic) is doing translation internally, so it feels wrong to require pre-translated strings here. @abcdefg30 @dragunoff what do you think?

Reply:
If this was discussed then it was probably discussed on Discord, but I don't remember any details either. Off the top of my head I'd guess that we didn't want to require this to not break all existing scripts (although I guess it could fall back to the literal parameter if it can't be parsed as translation key and log a warning)?

Reply:
The chrome code uses the literal value without logging any warnings (which IMO is the correct behaviour, for now at least).

Reply:
`Media.DisplayMessage` at the end is handled by `TextNotificationsManager.AddTextNotification` which doesn't do translations. And I think it's good to keep the API between these methods consistent. So the question maybe is do we want `TextNotificationsManager` to be aware of translations or just print whatever it's given? Moving the translation handling deeper would simplify the calling code a lot and we can always just fall back to the literal value if a translation is not found. And missing translations should be caught by the linter.

Reply:
My view is that translation support should be a core part of the engine, so should be handled as deeply/centrally as is feasible to avoid duplication.

Reply:
I doubt that would work with translation attribute path through.

Reply:
Maybe we need a new `Media.DisplayTranslatedMessage` which takes translation arguments then?

Reply:
I prefer to keep the standard we established in #20265.

Reply:
How about `Media.DisplayMessage` for no-arg translations and `Media.DisplayFormat` for translations with args?

Reply:
> I prefer to keep the standard we established in #20265.

It might make sense to take this as is, and then potentially change both mods in one go.

Reply:
I experimented a bit because I have to admit less boilerplate in scripting is a noble goal, however, causing a lot of internal and external API changes. `TextNotificationsManager` is not at all translation aware yet. This is a project worth its own pull request, maybe more.

Reply:
I think we could get away for now with doing the translation in `MediaGlobal`, and then moving it later into `TextNotificationsManager`. This gives us all the boilerplate and performance benefits without all the internal API changes.

I could live with merging this and cleaning up later, but IMO that clean up should be a hard blocker for Next + 1.

Reply:
We want text notifications to be localized, so this is going to happen anyway. I just wasn't aware of this possible connection. We just need to make them accept translation arguments and later glue the Lua table to C# dictionary to it.